### PR TITLE
[DST-379]: Docs: Persist Theme

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**

### CODE REVIEW

File: docs/ui/ThemeSwitch.tsx [hunk 1]
```diff
@@ -33,8 +42,43 @@ export const MarigoldThemeSwitch = ({
...
+
+      return;
+    }
+
+    setTheme(theme);
+    sessionStorage.setItem('theme', theme);
+    router.push(`?theme=${theme}`, {
+      scroll: false,
+    });
+  }, [theme]);
```
Comment (sarahgm) [inline docs/ui/ThemeSwitch.tsx:81]: there is a warning here in this file: `React Hook useEffect has missing dependencies: 'localTheme', 'router', and 'themeQueryParam'. Either include them or remove the dependency array`

Comment (aromko) [reply docs/ui/ThemeSwitch.tsx]: should be fixed now

**Comment by sebald** [comment]
LGTM just one question: with this implementation, people can not deeplink (with a certain theme). Should we add this or leave it?

@sarahgm @OsamaAbdellateef @aromko @KadaStrophe

File: docs/app/layout.tsx [hunk 2]
```diff
@@ -46,35 +47,34 @@ const Layout = ({ children }: { children: React.ReactNode }) => {
...
-          </MarigoldProvider>
-        </MarigoldThemeSwitch>
+        <Suspense>
+          <MarigoldThemeSwitch themes={themes} initial="b2b">
+            <MarigoldProvider theme={theme}>
+              <div className="flex min-h-screen flex-col">
+                <SiteHeader />
+                <div className="flex-1 px-4 lg:px-10 xl:px-20">
+                  <aside
+                    className={[
```
Comment (sebald) [inline docs/app/layout.tsx:57]: this overrides styles!

**Comment by OsamaAbdellateef** [comment]
> LGTM just one question: with this implementation, people can not deeplink (with a certain theme). Should we add this or leave it?
> 
> @sarahgm @OsamaAbdellateef @aromko @KadaStrophe

you mean when you have a deep link and then switch the theme u lose the deep link  ? @sebald

**Review by sebald — CHANGES_REQUESTED** [review]

**Comment by OsamaAbdellateef** [comment]
> > > LGTM just one question: with this implementation, people can not deeplink (with a certain theme). Should we add this or leave it?
> > > @sarahgm @OsamaAbdellateef @aromko @KadaStrophe
> > 
> > 
> > you mean when you have a deep link and then switch the theme u lose the deep link ? @sebald
> 
> Yes :) like always having the `?theme=core` query appended

For me i would say it is good to have

**Comment by OsamaAbdellateef** [comment]
> > > LGTM just one question: with this implementation, people can not deeplink (with a certain theme). Should we add this or leave it?
> > > @sarahgm @OsamaAbdellateef @aromko @KadaStrophe
> > 
> > 
> > you mean when you have a deep link and then switch the theme u lose the deep link ? @sebald
> 
> Yes :) like always having the `?theme=core` query appended

done :heavy_check_mark:  @sebald

File: docs/ui/ThemeSwitch.tsx [hunk 3]
```diff
@@ -33,11 +42,55 @@ export const MarigoldThemeSwitch = ({
...
+  }
+
+  const isInitialMount = useRef(true); // Ref to track initial mount
+
+  const updateTheme = useCallback(
+    (theme: string) => {
+      setTheme(theme);
+      localStorage.setItem('theme', theme);
+
+      const currentUrl = new URL(window.location.href);
```
[the remark this answers is not in the document]
Comment (OsamaAbdellateef) [reply docs/ui/ThemeSwitch.tsx:64]: The point is that  hooks(`useSeachParams`) can't be used inside `useCallback` , and if it is used outside it won't get or see the changes in URL  @sebald

**Comment by OsamaAbdellateef** [comment]
Should be working now @sebald

**Review by sebald — APPROVED** [review]

**Pull request merged**